mono/metadata/unity-memory-info.c

```diff
@@ -24,67 +24,82 @@ typedef struct CollectMetadataContext
 	MonoMetadataSnapshot* metadata;
 } CollectMetadataContext;
 
-static void ContextInsertClass(CollectMetadataContext* context, MonoClass* klass)
+static void
+ContextRecurseClassData (CollectMetadataContext *context, MonoClass *klass)
 {
 	gpointer orig_key, value;
+	gpointer iter = NULL;
+	MonoClassField *field = NULL;
+	int fieldCount;
+
 	/* use g_hash_table_lookup_extended as it returns boolean to indicate if value was found.
 	* If we use g_hash_table_lookup it returns the value which we were comparing to NULL. The problem is
 	* that 0 is a valid class index and was confusing our logic.
 	*/
-	if (klass->inited && !g_hash_table_lookup_extended(context->allTypes, klass, &orig_key, &value))
-		g_hash_table_insert(context->allTypes, klass, GINT_TO_POINTER(context->currentIndex++));
+	if (!g_hash_table_lookup_extended (context->allTypes, klass, &orig_key, &value))
+		g_hash_table_insert (context->allTypes, klass, GINT_TO_POINTER (context->currentIndex++));
+
+	fieldCount = mono_class_num_fields (klass);
```

Comment:
Do we always want to do this, or only if the insert succeeds?

Reply:
oh ops only if it succeeds fixing up, also making sure we don't do the extra lookup there, fixed in cs 
90bf666